Core/include/Acts/EventData/Measurement.hpp

```diff
@@ -391,4 +399,232 @@ using FittableMeasurement =
+/// available before the measurement is accessed; e.g. the propagator provides
+/// the surface pointer during navigation, which is then used to lookup possible
+/// measurements.
+///
+/// The pointed-to geometry object would differ depending on the parameter type.
+/// This means either, that there needs to be an additional variable type or
+/// that a pointer to a base object is stored (requiring a `dynamic_cast` later
+/// on). Both variants add additional complications. Since the geometry object
+/// is not required anyways (as discussed above), not storing it removes all
+/// these complications altogether.
+template <typename source_link_t, typename indices_t, size_t kSize>
+class FixedSizeMeasurement {
+  static_assert(SourceLinkConcept<source_link_t>,
+                "Source link does not fulfill SourceLinkConcept");
+
+  static constexpr size_t kFullSize = detail::kParametersSize<indices_t>;
+
+  using Subspace = detail::FixedSizeSubspace<kFullSize, kSize>;
+
+ public:
+  using Scalar = typename detail::ParametersScalar<indices_t>;
+  /// Vector type containing for measured parameter values.
+  using ParametersVector = ActsVector<Scalar, kSize>;
+  /// Matrix type for the measurement covariance.
+  using CovarianceMatrix = ActsSymMatrix<Scalar, kSize>;
+  /// Vector type containing all parameters in the same space.
+  using FullParametersVector = ActsVector<Scalar, kFullSize>;
+  using ProjectionMatrix = ActsMatrix<Scalar, kSize, kFullSize>;
+  using ExpansionMatrix = ActsMatrix<Scalar, kFullSize, kSize>;
+
+  /// Construct from source link, subset indices, and measured data.
+  ///
+  /// @tparam parameters_t Input parameters vector type
+  /// @tparam covariance_t Input covariance matrix type
+  /// @param source The link that connects to the underlying detector readout
+  /// @param indices Which parameters are measured
+  /// @param params Measured parameters values
+  /// @param cov Measured parameters covariance
+  ///
+  /// @note The indices must be ordered and must describe/match the content
+  ///   of parameters and covariance.
+  template <typename parameters_t, typename covariance_t>
+  FixedSizeMeasurement(source_link_t source,
+                       const std::array<indices_t, kSize>& indices,
+                       const Eigen::MatrixBase<parameters_t>& params,
+                       const Eigen::MatrixBase<covariance_t>& cov)
+      : m_source(std::move(source)),
+        m_subspace(indices),
+        m_params(params),
+        m_cov(cov) {
+    // TODO we should be able to support arbitrary ordering, by sorting the
+    //   indices and reordering parameters/covariance. since the parameter order
+    //   can be modified by the user, the user can not always know what the
+    //   right order is. another option is too remove the requirement for index
+    //   ordering from the subspace types, but that will make it harder to
+    //   refactor their implementation later on.
```

Comment:
Right, can we not make a `integer_sequence` out of the indices and then reorder them by that?

Reply:
This can only happen at runtime; so this can not be handled with an `integer_sequence` at compile time. We also need to reorder the parameters/covariance content. This is not something that is a problem at the moment, but I wanted to avoid that this gets forgotten.

Reply:
I don't have a problem with this being here, but we could think about moving the prose to a doc page and link to that from the comment here.

Reply:
This is a `TODO`, not documentation. It might be a good idea for the other comments. I need to rewrite the EDM documentation anyways, but will only do so after the changes are in.